src/main/java/fi/okm/jod/yksilo/controller/profiili/KoulutusController.java

```diff
@@ -42,41 +42,65 @@ class KoulutusController {
   private final KoulutusService service;
 
   @GetMapping
-  List<KoulutusDto> getAll(@AuthenticationPrincipal JodUser user) {
-    return service.findAll(user);
+  List<KoulutusKategoriaDto> find(
+      @AuthenticationPrincipal JodUser user, @RequestParam(required = false) String kategoriaId) {
+    return switch (kategoriaId) {
+      case null -> service.findAll(user);
+      case "null" -> service.findAll(user, null);
```

Comment:
Is this string expected to be "null" instead of "" empty at some point?

Reply:
We need to somehow distinguish "fetch all Koulutus" and "find all Koulutus without a Kategoria". Using "null" here is probably not optimal, but maybe more explicit than `GET /api/koulutukset?kategoriaId=`